### PR TITLE
feat(marriage-conditions): Marriage conditions application template

### PR DESCRIPTION
**Pull request by albinagu**
Marriage Conditions template WIP behind feature flag

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [x] I have rebased against main before asking for a review

### CODE REVIEW

File: libs/application/templates/marriage-conditions/src/fields/ApplicationOverview/ApplicationOverview.tsx
Comment (kksteini) [inline libs/application/templates/marriage-conditions/src/fields/ApplicationOverview/ApplicationOverview.tsx:13]: :eyes: :eyes: :eyes:

File: libs/application/templates/marriage-conditions/src/fields/ApplicationOverview/ApplicationOverview.tsx
Comment (kksteini) [inline libs/application/templates/marriage-conditions/src/fields/ApplicationOverview/ApplicationOverview.tsx:70]: Hard coded string?

File: libs/application/templates/marriage-conditions/src/fields/InfoAlert/InfoAlert.tsx
Comment (kksteini) [inline libs/application/templates/marriage-conditions/src/fields/InfoAlert/InfoAlert.tsx:13]: Ditto, hard coded string

File: libs/application/templates/marriage-conditions/src/fields/NationalIdWithName/NationalIdWithName.tsx
Comment (kksteini) [inline libs/application/templates/marriage-conditions/src/fields/NationalIdWithName/NationalIdWithName.tsx:80]: :detective: Hard code?

File: libs/application/templates/marriage-conditions/src/lib/constants.ts
Comment (kksteini) [inline libs/application/templates/marriage-conditions/src/lib/constants.ts:41]: Not a review comment but just curious. Is this really not accessible from an API?

File: libs/application/templates/marriage-conditions/src/lib/dataSchema.ts
Comment (kksteini) [inline libs/application/templates/marriage-conditions/src/lib/dataSchema.ts:7]: There is `zod.string().email()` I believe.
Not a review comment since there may be cases where `refine` is preferred.

**Review by kksteini — APPROVED** [review]

File: libs/application/templates/marriage-conditions/src/dataProviders/NationalRegistryMaritalStatusProvider.ts
Comment (alexdiljar) [reply libs/application/templates/marriage-conditions/src/dataProviders/NationalRegistryMaritalStatusProvider.ts:15]: Our thought process was keeping this here for now, and then once we start moving `dataProviders` to `template-api-modules `that a `nationalRegistryXroadUser` provider should be added. That would contain the marital status so that everyone can use it. I am a bit concerned that all applications are using the national registry soap client to get info about applicants.

File: libs/application/templates/marriage-conditions/src/fields/Done/SpouseDone.tsx
Comment (Thorrik58) [inline libs/application/templates/marriage-conditions/src/fields/Done/SpouseDone.tsx:17]: For the sake of avoiding custom fields, could this have been solved with a description field with markdown?

File: libs/application/templates/marriage-conditions/src/forms/done.ts
Comment (Thorrik58) [inline libs/application/templates/marriage-conditions/src/forms/done.ts:24]: should the redirect to minar sidur be added here?

File: libs/application/templates/marriage-conditions/src/fields/NationalIdWithName/NationalIdWithName.tsx
Comment (Thorrik58) [inline libs/application/templates/marriage-conditions/src/fields/NationalIdWithName/NationalIdWithName.tsx:14]: I think you can add this query in a queries section of the application and use the codegen to generate easy to use hooks for you. But this works.

File: libs/clients/syslumenn/src/lib/syslumennClient.service.ts
Comment (RunarVestmann) [inline libs/clients/syslumenn/src/lib/syslumennClient.service.ts:185]: Maybe this console.log isn't intentionally left here :)

**Review by RunarVestmann — APPROVED** [review]
Looks alright, Stefna is only a code owner of one file here (syslumennClient.service.ts) and there's a console.log there with maybe isn't intentionally left there. I'll approve though

File: libs/application/template-api-modules/src/lib/modules/templates/marriage-conditions-submission/emailGenerators/assignOtherSpouseEmail.ts
Comment (alexdiljar) [reply libs/application/template-api-modules/src/lib/modules/templates/marriage-conditions-submission/emailGenerators/assignOtherSpouseEmail.ts:38]: Do you know of any examples of this in the code?

Comment (alexdiljar) [reply libs/application/template-api-modules/src/lib/modules/templates/marriage-conditions-submission/emailGenerators/assignOtherSpouseEmail.ts:38]: I can add it in hard coded, would be nice to have this in the `emailOptions` in `BaseTemplateAPIModuleConfig` so that all future applications can be using the same replyTo `umsoknir@island.is`. I'll add it as a task on the application task list in Notion :)

**Review by Thorrik58 — APPROVED** [review]
LGTM

**Review by Snojo — APPROVED** [review]
Looking good on devops

File: libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts
Comment (saevarma) [inline libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts:97]: You can return the promise directly without awaiting.

File: libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts
Comment (saevarma) [inline libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts:111]: Ditto on return await.

**Review by saevarma — APPROVED** [review]
Core files LGTM.
Two minor comments on return await. I removed the `automerge` label so you can review it if you want to address it before merging.

File: libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts
Comment (alexdiljar) [reply libs/api/domains/national-registry-x-road/src/lib/nationalRegistryXRoad.resolver.ts:111]: There are return awaits on most of the endpoints in this resolver, I'll remove them all then.

**Pull request merged**
